MekHQ/src/mekhq/campaign/universe/RATGeneratorConnector.java

```diff
@@ -53,126 +52,124 @@ public RATGeneratorConnector(int year) {
+     * We force a fallback to try to ensure that something is generated if the parents have any possible units to generate,
+     * as that is the normally expected behaviour for MekHQ OpFor generation.
      * @param count How many units to generate
      * @param parameters RATGenerator parameters
      */
     @Override
-    public List<MechSummary> generate(int count, UnitGeneratorParameters parameters) {
-        UnitTable ut = UnitTable.findTable(parameters.getRATGeneratorParameters());
-
-        return ut == null ? new ArrayList<MechSummary>() :
-            ut.generateUnits(count, parameters.getFilter() == null ? null : ms -> parameters.getFilter().test(ms));
+    public List<MechSummary> generate(final int count, final UnitGeneratorParameters parameters) {
+        final UnitTable table = findOpForTable(parameters);
+        return table.generateUnits(count, (parameters.getFilter() == null) ? null : ms -> parameters.getFilter().test(ms));
     }
 
     /**
      * Generates a single mech summary from a RAT determined by the given faction, quality and other parameters.
-     * Note that for the purposes of this implementation, any faction record or unit quality are ignored.
+     * We force a fallback to try to ensure that something is generated if the parents have any possible units to generate,
+     * as that is the normally expected behaviour for MekHQ OpFor generation.
      * @param parameters RATGenerator parameters
      */
     @Override
-    public MechSummary generate(UnitGeneratorParameters parameters) {
-        UnitTable ut = UnitTable.findTable(parameters.getRATGeneratorParameters());
+    public @Nullable MechSummary generate(final UnitGeneratorParameters parameters) {
+        final UnitTable table = findOpForTable(parameters);
+        return table.generateUnit((parameters.getFilter() == null) ? null : ms -> parameters.getFilter().test(ms));
+    }
 
-        return ut == null ? null :
-            ut.generateUnit(parameters.getFilter() == null ? null : ms -> parameters.getFilter().test(ms));
+    /**
+     * This finds a unit table for OpFor generation. It falls back using the parent faction to try to ensure there are
+     * units in the unit table, so an OpFor is generated.
+     * @param unitParameters the base parameters to find the table using.
+     * @return the unit table to generate using
+     */
+    private UnitTable findOpForTable(final UnitGeneratorParameters unitParameters) {
+        final UnitTable.Parameters parameters = unitParameters.getRATGeneratorParameters();
+        UnitTable table = UnitTable.findTable(parameters);
+        if (!table.hasUnits()) {
+            // Do Parent Factions Fallbacks to try to ensure units can be generated, at a maximum of 10
+            List<String> factions = parameters.getFaction().getParentFactions();
+            for (int i = 0; (i < 10) && !factions.isEmpty(); i++) {
+                final List<String> parentFactions = new ArrayList<>();
```

Comment:
Avoid potential dupes perhaps? Not sure the incidence of this.
```suggestion
                final Set<String> parentFactions = new HashSet<>();
```

Reply:
A set is probably a better choice, will swapover in the morrow